src/include/daos_array.h

```diff
@@ -19,6 +19,10 @@
 extern "C" {
 #endif
 
+#include <daos_types.h>
+#include <daos_obj.h>
+#include <daos_obj_class.h>
+
```

Comment:
im not going to comment again on every one of those public headers and will just comment on his one.
so the idea is for user to include daos.h, and that pulls in all what's needed. so "fixing" all the headers to be able to be included by itself is not required.